src/Fixer/LanguageConstruct/SingleSpaceAfterConstructFixer.php

```diff
@@ -0,0 +1,220 @@
+    {
+        parent::configure($configuration);
+
+        if (70000 <= \PHP_VERSION_ID) {
+            self::$tokenMap['yield_from'] = T_YIELD_FROM;
+        }
+
+        $this->fixTokenMap = [];
+
+        foreach ($this->configuration['constructs'] as $key) {
+            $this->fixTokenMap[$key] = self::$tokenMap[$key];
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Ensures a single space after language constructs.',
+            [
+                new CodeSample(
+                    '<?php
+
+throw  new  \Exception();
+'
+                ),
+                new CodeSample(
+                    '<?php
+
+echo  "Hello!";
+',
+                    [
+                        'constructs' => [
+                            'echo',
+                        ],
+                    ]
+                ),
+                new VersionSpecificCodeSample(
+                    '<?php
+
+yield  from  baz();
+',
+                    new VersionSpecification(70000),
+                    [
+                        'constructs' => [
+                            'yield_from',
+                        ],
+                    ]
+                ),
+            ]
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isAnyTokenKindsFound(array_values($this->fixTokenMap));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        $tokenKinds = array_values($this->fixTokenMap);
+
+        for ($index = $tokens->count() - 1; $index >= 0; --$index) {
+            $token = $tokens[$index];
+
+            if (!$token->isGivenKind($tokenKinds)) {
+                continue;
+            }
+
+            $whitespaceTokenIndex = $index + 1;
+
+            if (';' === $tokens[$whitespaceTokenIndex]->getContent()) {
+                continue;
+            }
+
+            if (
+                $token->isGivenKind(T_STATIC)
+                && !$tokens[$tokens->getNextMeaningfulToken($index)]->isGivenKind([T_FUNCTION, T_VARIABLE])
+            ) {
+                continue;
+            }
+
+            if (!$tokens[$whitespaceTokenIndex]->equals([T_WHITESPACE])) {
+                $tokens->insertAt($whitespaceTokenIndex, new Token([T_WHITESPACE, ' ']));
+            } elseif (' ' !== $tokens[$whitespaceTokenIndex]->getContent()) {
+                $tokens->clearAt($whitespaceTokenIndex);
+                $tokens->insertAt($whitespaceTokenIndex, new Token([T_WHITESPACE, ' ']));
+            }
+
+            if (
+                70000 <= \PHP_VERSION_ID
+                && $token->isGivenKind(T_YIELD_FROM)
+                && 'yield from' !== strtolower($token->getContent())
+                && 1 === Preg::match('/^yield\s+from$/i', $token->getContent())
```

Comment:
this will always be true, otherwise it is not Tokenized as T_YIELD_FROM so for performance the check can be removed